include/caffe/parallel.hpp

```diff
@@ -14,6 +15,12 @@
 #include "caffe/syncedmem.hpp"
 #include "caffe/util/blocking_queue.hpp"
 
+#ifdef USE_NCCL
```

Comment:
Yes we could guard the whole file and the corresponding sections in caffe.cpp and python.